docs/operating-eck/configure-operator-metrics.asciidoc

```diff
@@ -0,0 +1,356 @@
+
+[source,shell,subs="attributes,+macros"]
+----
+kubectl patch sts -n elastic-system elastic-operator --patch-file=/dev/stdin <<-EOF
+spec:
+  template:
+    spec:
+      containers:
+        - name: manager
+          ports:
+          - containerPort: 9443
+            name: https-webhook
+            protocol: TCP
+          - containerPort: 8080
+            protocol: TCP
+            name: metrics
+EOF
+----
+
+* Restart the ECK Operator.
+
+[source,sh]
+----
+kubectl delete pod -n elastic-system elastic-operator-0
+----
+
+* Install the `PodMonitor` to allow scraping of the metrics endpoint by Prometheus.
+
+[source,shell,subs="attributes,+macros"]
+----
+cat $$<<$$EOF | kubectl apply -f -
+apiVersion: monitoring.coreos.com/v1
+kind: PodMonitor
+metadata:
+  name: elastic-operator
+  namespace: elastic-system
+  labels:
+    control-plane: elastic-operator
+    app.kubernetes.io/component: metrics
+spec:
+  podMetricsEndpoints:
+    - port: metrics
+      path: /metrics
+      interval: 1m
+      scrapeTimeout: 30s
+  namespaceSelector:
+    matchNames:
+      - elastic-system
+  selector:
+    matchLabels:
+      control-plane: elastic-operator
+EOF
+----
+
+== Securing the ECK Operator metrics endpoint
+
+NOTE: The ECK Operator metrics endpoint will be secured by default beginning in version 2.14.0
+
+The ECK Operator provides a metrics endpoint that can be used to monitor the operator's performance and health. By default, the metrics endpoint is not enabled and is not secured. To enable the metrics endpoint follow the previous instructions. To enable TLS/HTTPs and RBAC on the metrics endpoint, follow the instructions in the following sections depending on whether you installed ECK through the Helm chart or the manifests.
+
+=== Using the ECK Operator Helm Chart
+
+If you installed ECK through the Helm chart commands listed in <<{p}-install-helm>>, you can now set `config.metrics.secureMode.enable` to `true` and both RBAC and TLS/HTTPs will be enabled for the metrics endpoint.
+
+==== Using your own TLS certificate for the metrics endpoint when using the Helm Chart
+
+By default a self-signed certificate will be generated for use by the metrics endpoint. If you want to use your own TLS certificate for the metrics endpoint you can provide the `config.metrics.secureMode.tlsCertSecret` to the Helm chart. The `tlsCertSecret` should be the name of a Kubernetes `Secret` that contains the TLS certificate. The same secret can also contain an optional TLS key, and if so provide the same secret name to `config.metrics.secureMode.tlsPrivateKeySecret`. The following keys are supported within the secret:
+
+* `tls.crt` - The PEM-encoded TLS certificate
+* `tls.key` - (Optional) The PEM-encoded TLS private key
+
+Using your own TLS certificate will still enable `tlsConfig.insecureSkipVerify` within the `ServiceMonitor` to allow Prometheus to scrape the metrics endpoint. This setting may be revisited in the future.
```

Comment:
Should this be configurable?

Reply:
I discuss this a bit in the *Note* section of the initial description of this PR @pebrc . I was definitely wanting some feedback on this as I was a bit conflicted.

Reply:
The main reason I was wondering about this was that this sentence in the documenation seemed a bit odd. Could we not add a Helm value that allows users to turn this off but default to `insecureSkipVerify`

Reply:
All updated, this should be good for reviews.